openedx/core/djangoapps/theming/templatetags/optional_include.py

```diff
@@ -0,0 +1,77 @@
+"""
+The functions in this module are based on the contents of
+https://github.com/django/django/blob/1.8.5/django/template/loader_tags.py --
+specifically, the do_include function. It has been modified as little as
+possible, in order to match the behavior of the {% include %} template tag,
+except for making it optional.
+"""
+# Because we want to match the original loader_tags.py file as closely as
+# possible, we should disable pylint so it doesn't complain about the violations
+# that are already in that file
+# pylint: skip-file
+from django.template.base import (
+    TemplateSyntaxError, Library, token_kwargs, TemplateDoesNotExist
+)
+from django.template.loader_tags import IncludeNode
+
+register = Library()
+
+
+class OptionalIncludeNode(IncludeNode):
+    def render(self, context):
+        try:
+            super(OptionalIncludeNode, self).render(context)
+        except TemplateDoesNotExist:
+            return ''
+
+
+@register.tag('optional_include')
```

Comment:
This is a copy of 50 lines of code from Django. I understand why you had to do it, but maybe we should put a big ackknowledgment of it here or something?  We've taken on the maintenance burden here.

I'm also torn between keeping all the behavior of the original, and stripping it down to just the parts we need.